explain using PTFs in SUSE Manager / Uyuni

mcalmer opened the pull request:
# Description

Project Lotus adds the functionality to use PTF in Uyuni. Describe how to do it.

SMEs: @mackdk @mcalmer 

# Target branches

Which documentation version does this PR apply to?

- [x] Master (Default)
- [x] Manager-4.3 when PR is merged https://github.com/SUSE/spacewalk/pull/19933
- [ ] Manager-4.2



# Links

Fixes https://github.com/SUSE/spacewalk/issues/20322

mackdk commented on modules/administration/pages/ptfs.adoc:
```diff
@@ -0,0 +1,81 @@
+[[ptfs]]
+= Using PTFs in {productname}
+
+SUSE provide for their Enterprise Products temporary fixes shipped to individual customers.
+These PTFs (Program Temporary Fixes) can now be made available as repositories which can be synced in {productname}.
```
```suggestion
SUSE provides temporary fixes for all currently supported solutions shipped directly to its customers.
These PTFs (Program Temporary Fixes) are now available as repositories, which can be synced in {productname}.
```

mcalmer replied:
used that. thanks

mackdk commented on modules/administration/pages/ptfs.adoc:
```diff
@@ -0,0 +1,81 @@
+== Understanding PTF packages
+
+PTF packages are installed via a proxy package and are named `ptf-xxxxxx`.
+They will depend on the correct version of the package that is known to include the correction in the software. This type of package
```
```suggestion
They will depend on the correct version of the package that is known to include the software correction. This type of package:
```

mackdk commented on modules/administration/pages/ptfs.adoc:
```diff
@@ -0,0 +1,81 @@
+PTF packages are installed via a proxy package and are named `ptf-xxxxxx`.
+They will depend on the correct version of the package that is known to include the correction in the software. This type of package
+
+* cannot be installed accidentally (ie. zypper update will never suggest installing them),
+* cannot be removed accidentally (ie. a newer package version will not replace the PTF one, unless the user makes it explicitly on the zypper command line),
+* is only updated when the newer version is known to address the specific issue addressed first by the PTF,
+* updates only packages already installed on the system (a software that is split in multiple packages will have only the packages currently installed on the system being replaced).
```
```suggestion
* cannot be installed accidentally (i.e. zypper update will never suggest installing them),
* cannot be removed accidentally (i.e. a newer package version will not replace the PTF one, unless the user makes it explicitly on the zypper command line),
* is only updated when the newer version is known to address the specific issue previously solved by the PTF,
* updates only packages already installed on the system (i.e. if a software is split into multiple packages, the PTF will replace only those currently installed on the system).
```

mcalmer replied:
This is a 1:1 copy of the TID. I will let our doc team correct this or not

admd commented on modules/administration/pages/ptfs.adoc:
```diff
@@ -0,0 +1,81 @@
+[[ptfs]]
+= Using PTFs in {productname}
+
+SUSE provide for their Enterprise Products temporary fixes shipped to individual customers.
```
```suggestion
SUSE provides for their Enterprise Products temporary fixes shipped to individual customers.
```

admd reviewed (commented):
Overall looks good, just a couple of suggestion

1. Anything to be noted down about mixed env where people are installing some PTFs(e.g mass ones) in traditional way and then in some case though this new way?
2. Some people do manual stuff from terminal, would be good to add some info about `zypper removeptf `command
3. In case new MU is released with the fix, can/do we show in UI that these installed PTFs can be removed now safely or not needed?
4. Do we provide API too? if yes, we should add a note here as well.
5.  We could already add a note about syncing optional channel from UI as that part will be done till then.

mackdk commented on modules/administration/pages/ptfs.adoc:
```diff
@@ -0,0 +1,81 @@
+
+Now this channel is available and can be added to every system which is using the same base channel.
+
+PTF packages needs to be installed explicit. They are not automatically installed when updating a system.
+The SUSE Customer Support will tell the ptf number to fix a specific problem.
+In {productname} {webui} every system with PTFs available for installation has a page which list them.
```
```suggestion
PTF packages need to be installed explicitly, since they are not automatically picked up when updating a system.
The SUSE Customer Support will provide the PTF number to fix a specific problem.
In {productname} {webui} every system with PTFs available for installation has a page which list them.
```

mcalmer replied:
used it

mackdk commented on modules/administration/pages/ptfs.adoc:
```diff
@@ -0,0 +1,81 @@
+
+.Procedure: Installation of PTF packages
+
+. In the {productname} {webui}, navigate to menu:Systems[Systems List] and select the client you want to install a PTF.
```
```suggestion
. In the {productname} {webui}, navigate to menu:Systems[Systems List] and select the client where you want to install a PTF.
```

mcalmer replied:
:+1:

mackdk reviewed (commented):
Looks good to me. I added some minor rewording proposal. Feel free to integrate them if you think they simplify the text.

mcalmer commented:
> Overall looks good, just a couple of suggestion
> 
> 1. Anything to be noted down about mixed env where people are installing some PTFs(e.g mass ones) in traditional way and then in some case though this new way?

No idea how to do this. The text is getting too complex and confusing. I think the customer should decide what he use.

> 2. Some people do manual stuff from terminal, would be good to add some info about `zypper removeptf `command

added

> 3. In case new MU is released with the fix, can/do we show in UI that these installed PTFs can be removed now safely or not needed?

I think this is not needed. A normal update will replace it when a MU will have a newer version. I added that to the text.

> 4. Do we provide API too? if yes, we should add a note here as well.

The normal package API can be used. I added this to the text.

> 5. We could already add a note about syncing optional channel from UI as that part will be done till then.

Yes, but I think @parlt91 should add this. Pascal: please add a section about enabling optional channels via Web UI to this draft.

keichwa commented on modules/administration/pages/ptfs.adoc:
```diff
@@ -0,0 +1,103 @@
+. To see PTF installation results, navigate to menu:Systems[Events > History].
+
+
+In case a PTF should be installed using the API, the normal [systemitem]``system.schedulePackageInstall`` api can be used with the proxy package name.
```
```suggestion
In case a PTF should be installed using the API, the normal [systemitem]``system.schedulePackageInstall`` API can be used with the proxy package name.
```

keichwa commented on modules/administration/pages/ptfs.adoc:
```diff
@@ -0,0 +1,103 @@
+a simple package remove cannot be used. The ptf package is not selectable in the standard package list page.
+
+. In the {productname} {webui}, navigate to menu:Systems[Systems List] and select the client you want to remove a PTF from.
+. Navigate to the menu:Systems[Software > PTFs > List/Remove] sub tab
+. Select the ptf package you want to remove
```
```suggestion
. Navigate to the menu:Systems[Software > PTFs > List/Remove] sub tab.
. Select the ptf package you want to remove.
```

keichwa commented on modules/administration/pages/ptfs.adoc:
```diff
@@ -0,0 +1,103 @@
+
+== Removing the patched version of a package on the client
+
+In case a PTF should be removed directly on the client using the console, it is required to use a special
+command ``zypper removeptf ...``. All other ways result either in an error or can lead to unwanted
+behavior like removing very importand packages from the system and make the system unusable.
```
```suggestion
In case a PTF should be removed directly on the client using the console, it is required to use a special command ``zypper removeptf ...``.
  All other ways result either in an error or can lead to unwanted behavior like removing important packages from the system and make the system unusable.
```

keichwa reviewed (approved):
Some formal change proposals.
When it is merged we can propose more cosmetic changes according to our styleguide.

ktsamis commented on modules/administration/pages/ptfs.adoc:
```diff
@@ -0,0 +1,103 @@
+
+== Understanding PTF packages
+
+PTF packages are installed via a proxy package and are named `ptf-xxxxxx`.
```
```suggestion
PTF packages are installed via a proxy package and are named `ptf-xxxxxx`, where xxxxxx is a number and part of the name of the package, not its version.

```

ktsamis commented on modules/administration/pages/ptfs.adoc:
```diff
@@ -0,0 +1,103 @@
+When this regular update with the fix is released, an updated version of the PTF will also be released into
+the account-specific PTF repository. The updated PTF will remove the strict dependencies and allow updates to be installed again.
+
+The replacement of the PTF with the maintenance update which include the fix happens automatically via
```
```suggestion
The replacement of the PTF with the maintenance update which includes the fix happens automatically via
```

ktsamis commented on modules/administration/pages/ptfs.adoc:
```diff
@@ -0,0 +1,103 @@
+== Removing the patched version of a package on the client
+
+In case a PTF should be removed directly on the client using the console, it is required to use a special command ``zypper removeptf ...``.
+  All other ways result either in an error or can lead to unwanted behavior like removing important packages from the system and make the system unusable.
```
```suggestion
All other ways result either in an error or can lead to unwanted behavior like removing important packages from the system and make the system unusable.
```

ktsamis reviewed (commented):
Small typos and a small clarification, hopefully it helps :)

keichwa commented:
I agree with @ktsamis suggestions and fixes, thanks!

keichwa reviewed (approved):
Before merging we probably must run the l10n-weblate script because here we add a new file.  @jcayouette

mcalmer commented:
Removing "Draft" as now also the doc from Pascal is in about adding PTFs repos via UI.
